core/src/main/scala/org/apache/spark/util/collection/ExternalAppendOnlyMap.scala

```diff
@@ -126,7 +131,11 @@ class ExternalAppendOnlyMap[K, V, C](
 
     while (entries.hasNext) {
       curEntry = entries.next()
-      if (maybeSpill(currentMap, currentMap.estimateSize())) {
+      val estimatedSize = currentMap.estimateSize()
```

Comment:
Won't this be really expensive? It looks like this will try to re-estimate the map's size on each insertion. Does the map make optimizations so that this is cheap?